metricbeat/internal/metrics/opt.go

```diff
@@ -40,6 +42,27 @@ func OptUintWith(i uint64) OptUint {
 	}
 }
 
+// None marks the Uint as not having a value.
+func (opt *OptUint) None() {
+	opt.exists = false
+}
+
+// Exists returns true if the underlying value is valid
+func (opt OptUint) Exists() bool {
+	return opt.exists
+}
+
+// IsNone returns true if the value exists
+func (opt OptUint) IsNone(i uint64) {
+	opt.value = i
+	opt.exists = true
+}
```

Comment:
implementation and godoc do not match. Plus having another predicate on the type matching an already existing method adds redundancy. If possible interfaces should be kept small.